src/core/Form/TextInput/TextInput.test.tsx

```diff
@@ -142,30 +135,22 @@ describe('props', () => {
 
     describe('name', () => {
       const textInput = (
-        <TextInput
-          labelText="Test input"
-          name="test-name"
-          data-testid="input-name"
-        />
+        <TextInput labelText="Named test input" name="test-name" />
       );
-      const { getByTestId } = render(textInput);
-      const namedInput = getByTestId('input-name') as HTMLInputElement;
+      const { getByRole } = render(textInput);
+      const namedInput = getByRole('textbox', {
+        name: 'Named test input',
+      }) as HTMLInputElement;
 
       it('has the given name attribute', () => {
         expect(namedInput.name).toBe('test-name');
       });
     });
 
     describe('number', () => {
-      const textInput = (
-        <TextInput
-          labelText="Test input"
-          type="number"
-          data-testid="number-input"
-        />
-      );
-      const { getByTestId } = render(textInput);
-      const numberfield = getByTestId('number-input') as HTMLInputElement;
+      const textInput = <TextInput labelText="Number input" type="number" />;
+      const { getByRole } = render(textInput);
+      const numberfield = getByRole('spinbutton') as HTMLInputElement;
```

Comment:
This was new to me as well. Not obvious that an input type number is a spinbutton...